### PR TITLE
Call notifySessionError from native code

### LINKED ISSUE #3597: SyncSession#notifySessionError() seems to be unused
The comment for `SyncSession#notifySessionError()` says that it is called from `Session.cpp`, but I couldn't find `Session.cpp`.
There is a file `io_realm_internal_objectserver_SyncSession.cpp`, but it doesn't call the method.

If it's not needed, we should remove it.
If it should be called from JNI, we should fix it.

### PR DESCRIPTION
**Pull request by beeender**
Fix #3597

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 * Those were not kept by ProGuard: names of native methods not in the `io.realm.internal` package, names of classes used in method signature (#3596).
 * Missing ProGuard configuration for libraries used by Sync extension (#3596).
+* Fixed a bug that error handler is not called when sync session fails (#3597).
```
Comment (dalinaum) [inline CHANGELOG.md]: It looks like we usually omit `Fixed a bug ~`, so  is `error handler is not called when sync session fails (#3597).` enough?

File: realm/realm-library/src/main/cpp/objectserver_shared.hpp [hunk 2]
```diff
@@ -59,6 +59,11 @@ class JniSession {
             g_vm->AttachCurrentThread(&local_env, nullptr);
             std::string log = num_to_string(error_code) + " " + message.c_str();
             log_message(local_env, log_debug, log.c_str());
```
Comment (cmelchior) [inline realm/realm-library/src/main/cpp/objectserver_shared.hpp]: Do we want to log this here? The default java implementation also log these?

Comment (beeender) [inline realm/realm-library/src/main/cpp/objectserver_shared.hpp]: you are right, there is no need to log it here. It has been logged in the sync as well.

File: realm/realm-library/src/main/cpp/objectserver_shared.hpp [hunk 3]
```diff
@@ -54,11 +54,13 @@ class JniSession {
             }
         };
         auto error_handler = [&, global_obj_ref_tmp](int error_code, std::string message) {
-            // FIXME: Simplify this by moving log_message to AndroidLogger
             JNIEnv *local_env;
             g_vm->AttachCurrentThread(&local_env, nullptr);
-            std::string log = num_to_string(error_code) + " " + message.c_str();
-            log_message(local_env, log_debug, log.c_str());
+            jclass java_session_class = local_env->GetObjectClass(global_obj_ref_tmp);
```
Comment (kneth) [inline realm/realm-library/src/main/cpp/objectserver_shared.hpp]: Can `java_session_class` never be `NULL`?

Comment (beeender) [inline realm/realm-library/src/main/cpp/objectserver_shared.hpp]: never ... otherwise something is really wrong

Comment (bmunkholm) [inline realm/realm-library/src/main/cpp/objectserver_shared.hpp]: @beeender If it's allowed to return null we have to check for it. At least assert().
Also goes for the next method.

**Pull request merged**